src/styles/blocks/container.scss

```diff
@@ -0,0 +1,31 @@
+.container {
+  display: grid;
+  grid-template-columns: repeat(auto-fit, minmax($box-size, 1fr));
+
+  &__box {
+    height: $box-size;
+    display: flex;
+    justify-content: center;
+    align-items: center;
+    color: white;
+    font-size: 100px;
+
+    @for $boxNum from 1 through 6 {
+      &:nth-child(n + #{$boxNum})::before {
+        content: "#{$boxNum}";
+      }
+    }
+
+    @each $num, $color in $colors {
+      &:nth-child(n + #{$num}) {
+        background-color: $color;
+      }
+    }
+  }
+
+  @media (min-width: 1200px) {
+    & {
+      grid-template-columns: repeat(3, 1fr);
+    }
+  }
```

Comment:
No need to specify selector, write media query right under general rules for it.